### PR TITLE
feat(workout mirroring): `workoutSessionMirroringStartHandler` support + React Event emissions on `HKWorkoutSessionDelegate` methods

### PR DESCRIPTION
**Pull request by gentilijuanmanuel**
# Summary of changes

This PR adds support to two main `HealthKit` features that will enable us to make workout mirroring happen:
- `workoutSessionMirroringStartHandler`: method to set up the delegate that will react to remote workout session event updates. Official documentation can be found [here](https://developer.apple.com/documentation/healthkit/hkhealthstore/4172878-workoutsessionmirroringstarthand).
- `HKWorkoutSessionDelegate` methods:
1. `workoutSession(_:didChangeTo:from:date:)`: to react to workout session state changes (paused, resumed, ended). Official documentation can be found [here](https://developer.apple.com/documentation/healthkit/hkworkoutsessiondelegate/1627958-workoutsession).
2. `workoutSession(_:didFailWithError:)`: to react to possible errors that can happen during a workout session. Official documentation can be found [here](https://developer.apple.com/documentation/healthkit/hkworkoutsessiondelegate/1627951-workoutsession).
3. `workoutSession(_:didReceiveDataFromRemoteWorkoutSession:)`: to react to data that's coming from the workout session (like workout metrics). Official documentation can be found [here](https://developer.apple.com/documentation/healthkit/hkworkoutsessiondelegate/4266951-workoutsession).

Every time we get updates from delegate methods, we emit an event using the `EventEmitter`, and consumer apps like Runna will be able to listen to them and perform actions accordingly.

## Demo video

Video with our app as a consumer can be found in [this Slack message](https://runna-team.slack.com/archives/C035XF31W8P/p1731440247898019).

### CODE REVIEW

**Review by fnazarios — APPROVED** [review]
💯

File: src/utils/workoutSessionMirroringStartHandler.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import Native from '../native-types'
+
+const workoutSessionMirroringStartHandler = async () => Native.workoutSessionMirroringStartHandler()
```
Comment (walterholohan) [inline src/utils/workoutSessionMirroringStartHandler.ts:3]: Do we want to put any return types on this? 
```suggestion
const workoutSessionMirroringStartHandler: Promise<void> = async () => Native.workoutSessionMirroringStartHandler()
```

Comment (walterholohan) [reply src/utils/workoutSessionMirroringStartHandler.ts:3]: Does the function return anything?

**Review by walterholohan — APPROVED** [review]

File: src/utils/workoutSessionMirroringStartHandler.ts (hunk 1 above)
Comment (gentilijuanmanuel) [reply src/utils/workoutSessionMirroringStartHandler.ts]: The return type of this is: `() => Promise<boolean>`. Added it!

**Pull request merged**